### DISCUSSION TITLE
FEATURE: Add sourcegraph source

### DISCUSSION THREAD

**Discussion by ghostbuster91** [nvim-neo-tree/neo-tree.nvim#1245] [Ideas]
### Did you check the docs?

- [X] I have read all the docs.

### Is your feature request related to a problem? Please describe.

Not really a problem. I like to show tries in a consistent way no matter where they come from. 
I've just came across sourcegraph [plugin](https://github.com/sourcegraph/sg.nvim) for neovim and it seems to have a builtin file tree explorer. 

### Describe the solution you'd like.

Implement new source for sg.nvim. It should mimic the behavior of the sg's builtin file explorer. 

### Describe alternatives you've considered.

This can be a separate project, but I wanted to start somewhere and this repository seemed like a good choice.

### Additional Context

Video showing sg.nvim in use https://youtu.be/j5sfHG3z3ao?si=iiiiXYawQtYDcUv0&t=208

**Comment by miversen33** [comment]
While I am not the maintainer of Neo-tree, I would guess this is not something that is wanted as part of Core Neo-tree. That said, making it its own thing and adding it into Neo-tree is not hard at all. I believe there is an example repo floating around in the `nvim-neo-tree` org showing how to setup third party integration into Neo-tree but if there isn't you can always peek at [the neo-tree integration for netman.nvim](https://github.com/miversen33/netman.nvim/tree/main/lua/netman/ui/neo-tree)

**Comment by ghostbuster91** [comment]
Yes, I agree with you that it would be the best to make it a separate project. Just wanted to share this idea. Maybe a discussion would have been a better place. Feel free to move it there.

**Comment by cseickel** [comment]
Yes, an external source would be the best way to go about this. Here is the example repo: https://github.com/nvim-neo-tree/example-source

It is a very simple hello world. Netman or the internal filesystem source would be good real world examples of interactive lazy loaded file system browsing.

I'm going to convert this to a discussion.
